engine/src/main/java/org/pentaho/di/trans/steps/fileinput/text/TextFileInputReader.java

```diff
@@ -418,22 +412,10 @@ public void close() {
   }
 
   protected boolean tryToReadLine( boolean applyFilter ) throws KettleFileException {
-    String line;
-    line = TextFileInputUtils.getLine( log, isr, data.encodingType, data.fileFormatType, data.lineStringBuilder );
-
-    while ( line != null ) {
-        /*
-        Check that the number of enclosures in a line is even.
-        If not even it means that there was an enclosed line break.
-        We need to read the next line(s) to get the remaining data in this row.
-        */
-      if ( TextFileInputUtils.checkPattern( line, meta.getEnclosure() ) % 2 != 0 ) {
-        line = line + TextFileInputUtils.getLine( log, isr, data.encodingType, data.fileFormatType, data.lineStringBuilder );
-        lineNumberInFile++;
-      } else {
-        break;
-      }
-    }
+    String[] saGetLine =  TextFileInputUtils.getLine( log, isr, data.encodingType, data.fileFormatType, data.lineStringBuilder,
+      meta.getEnclosure(), lineNumberInFile );
+    String line = saGetLine[0];
+    lineNumberInFile = Long.parseLong( saGetLine[1] );
```

Comment:
I note below that this is a hack. This is an unnecessary data conversion which will create performance issues on large files.